include/dump_offload.hpp

```diff
@@ -0,0 +1,443 @@
+                    this->connection->sendStreamErrorStatus(
+                        boost::beast::http::status::internal_server_error);
+                    this->connection->close();
+                    return;
+                }
+                BMCWEB_LOG_CRITICAL("INFO: Hit Dump end of file");
+                this->connection->completionStatus = true;
+                this->connection->close();
+                return;
+            }
+
+            this->outputBuffer.commit(bytesRead);
+            auto streamHandler = [this, bytesRead, self(shared_from_this())]() {
+                this->outputBuffer.consume(bytesRead);
+                this->doReadStream();
+            };
+            this->connection->sendMessage(this->outputBuffer.data(),
+                                          streamHandler);
+        });
+    }
+
+    std::string entryID;
+    std::string dumpType;
+    boost::beast::flat_static_buffer<socketBufferSize> outputBuffer;
+    std::filesystem::path unixSocketPath;
+    boost::asio::local::stream_protocol::socket unixSocket;
+    uint64_t dumpSize{0};
+    boost::asio::steady_timer waitTimer;
+    crow::streaming_response::Connection* connection = nullptr;
+    uint16_t connectRetryCount{0};
+};
+
+static boost::container::flat_map<crow::streaming_response::Connection*,
+                                  std::shared_ptr<Handler>>
+    bmcHandlers;
+static boost::container::flat_map<crow::streaming_response::Connection*,
+                                  std::shared_ptr<Handler>>
+    systemHandlers;
+
+inline void resetHandlers()
+{
+    if (!systemHandlers.empty())
+    {
+        auto handler = systemHandlers.begin();
+        if (handler == systemHandlers.end())
+        {
+            BMCWEB_LOG_DEBUG("No handler to cleanup");
+            return;
+        }
+        if ((handler->second->dumpType == "system") ||
+            (handler->second->dumpType == "resource"))
+        {
+            handler->first->close();
+            BMCWEB_LOG_CRITICAL("INFO: {} dump resetHandlers cleanup",
+                                handler->second->dumpType);
+        }
+    }
+}
+
+inline void requestRoutes(App& app)
+{
+    BMCWEB_ROUTE(
+        app,
+        "/redfish/v1/Systems/system/LogServices/Dump/Entries/<str>/attachment/")
+        .privileges({{"ConfigureComponents", "ConfigureManager"}})
+        .streamingResponse()
+        .onopen([](crow::streaming_response::Connection& conn) {
+        if (!systemHandlers.empty())
+        {
+            BMCWEB_LOG_ERROR("Can't allow dump offload opertaion, one "
+                             "of the host dump is already offloading");
+            conn.sendStreamErrorStatus(
+                boost::beast::http::status::service_unavailable);
+            conn.close();
+            return;
+        }
+
+        std::string url(conn.req.target());
+        BMCWEB_LOG_CRITICAL("INFO:Request open for system dump offload with"
```

Comment:
Still needed?

Reply:
yeah, lets keep all these traces. since system dump offload failure will be a critical failure.
required to have all this data for debug

Reply:
remove this trace for now